chrome/content/zotero/elements/editableText.js

```diff
@@ -193,6 +193,9 @@
 				input.addEventListener('mousedown', this._handleMouseDown);
 				input.addEventListener('dragover', this._handleDragOver);
 				input.addEventListener('drop', this._handleDrop);
+				if (autocompleteEnabled) {
+					this.addEventListener('keydown', this._captureAutocompleteKeydown, true);
```

Comment:
Oh yeah. This could lead us to having multiple identical listeners on the `editable-text` which would not be good... Great catch!

To keep it clean, I wanted to avoid attaching a capturing listener (even one that does nothing) to non-autocomplete fields. My fear was that doing it could cause some confusion in the future. What if we add/remove that `_captureAutocompleteKeydown` in `autocomplete` setter? That way, it should be attached only once and also removed if `autocomplete` is cleared?

Reply:
Oh yeah, it actually doesn't add the same function multiple times. I didn't see it anywhere in the docs but that is certainly how it works. Cool, good to know!


I moved it back with a comment to clarify this thing a bit and removed it in `else` in case if autocomplete is removed.